orttraining/orttraining/training_ops/cpu/nn/dropout_op.h

```diff
@@ -10,27 +10,54 @@ namespace onnxruntime {
 namespace contrib {
 
 template <typename T1, typename T2>
-class Dropout final : public OpKernel {
+class Dropout : public OpKernel {
  public:
   Dropout(const OpKernelInfo& info) : OpKernel{info} {
     int64_t seed = 0;
     if (info.GetAttr<int64_t>("seed", &seed).IsOK()) {
       generator_ = onnxruntime::make_unique<RandomGenerator>(seed);
     }
+
+    trainable_dropout_ = false;
   }
 
   Status Compute(OpKernelContext* context) const override;
 
  private:
   mutable std::unique_ptr<RandomGenerator> generator_;
+
+ protected: 
+  bool trainable_dropout_;
```

Comment:
This can be a class template parameter... See softmax/logsoftmax for ref

This way, do you don't need to redefine another TrainableDropout Class, 
Dropout class can still have "final" property.

Reply:
This is a great idea! Thank you!